---
title: "PLK1 analysis of JHU Biobank data"
author:
- affiliation: Sage Bionetworks
  affiliation_url: https://sagebionetworks.org/
  name: Jineta Banerjee
  url: null
output:
  html_document:
    toc: yes
    number_sections: yes
    toc_float:
      collapsed: no
      smooth_scroll: no
    fig_width: 7
    fig_height: 6
    fig_caption: yes
    df_print: paged
  pdf_document:
    toc: yes
    always_allow_html: true
params:
  orig_date: 'Original Publish Date: Oct29, 2021'
  update_date: !r paste("Last updated on:", format(Sys.time(), '%d %B, %Y'))
description: |
  DEG analysis of JHU Biobank data
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
 
```{r lib_synapser, echo=FALSE, eval=TRUE, results='hide', message=FALSE, warning=FALSE}

library(BiocManager)
library(gProfileR)
library(tidyverse)
#library(glue)
library(edgeR)
library(limma)
library(pheatmap)
library(gprofiler2)

# library(DRUID)
# library(cauldron)
# library(AnnotationDbi)
# library(org.Hs.eg.db)
library(DT)

library(synapser)
synapser::synLogin(email= "jineta.banerjee",
                   apiKey= "iaIrESLwnfZef3H3+khpPhAtRrSgipl1YbFX9jAQ0YlL38ygCIZ2/ocleaivnCXX0Dzre81ahVPUjO79fLJgNg==")
```

## Publicly available data from JHU Biobank

We used the publicly available JHU Biobank RNASeq data stored in project : syn20812185 for this analysis.

```{r download data, eval=TRUE, echo=FALSE, results='hide', message=FALSE, warning=FALSE}


## All RNAseq data
NTAP_published_files <- synapser::synTableQuery(glue::glue("SELECT * FROM syn21221980"))$asDataFrame()

NTAP_pub_rnaseq_data <- synapser::synTableQuery(glue::glue("SELECT * FROM syn20812185"))$asDataFrame() %>% 
  dplyr::select(totalCounts, Symbol, zScore, specimenID, individualID, sex, tumorType, studyName)

### Clean up data
NTAP_pub_rnaseq_data$sex[NTAP_pub_rnaseq_data$sex == "female"] <- "Female"
NTAP_pub_rnaseq_data$sex[NTAP_pub_rnaseq_data$sex == "male"] <- "Male"
NTAP_pub_rnaseq_data$tumorType[NTAP_pub_rnaseq_data$tumorType == "Malignant peripheral nerve sheath tumor"] <- "Malignant Peripheral Nerve Sheath Tumor"

NTAP_pub_rnaseq_data <- NTAP_pub_rnaseq_data %>% 
  filter(tumorType %in% c("Cutaneous Neurofibroma", "Plexiform Neurofibroma", "Neurofibroma", "Malignant Peripheral Nerve Sheath Tumor"),
         !grepl('xenograft', specimenID, ignore.case = T),
         !grepl('Cell Line', specimenID, ignore.case = T),
         !specimenID %in% c("BI386-004","CW225-001","DW356-002",
                            "JK368-003", "SK436-005"))


```


We then used TMM normalization to prepare the dataset for differential gene expression analysis.

```{r TMM-LCPM normalization, eval=TRUE, echo=FALSE, results='hide', message=FALSE, warning=FALSE}

gene_mat<-reshape2::acast(unique(NTAP_pub_rnaseq_data),
                          Symbol~specimenID,
                          value.var='totalCounts',
                          fun.aggregate = mean)

# missing<-which(apply(gene_mat,1,function(x) any(is.na(x))))
# gene_mat<-gene_mat[-missing,]

# Remove low expressed genes
#print("Genes that passed low expression filter:")
keepRows <- rowSums((gene_mat[,]) >= 10) >= (ncol(gene_mat[,])*0.8)
table(keepRows)
gene_mat <- gene_mat[keepRows,]
# 
# ## avg expression filter
# keepRows <- apply(gene_mat[],1,mean) >= 1
# table(keepRows)
# gene_mat <- gene_mat[keepRows,]

DGE.all <-DGEList(counts=gene_mat) # function from edgeR

# TMM normalization
DGE.all <- calcNormFactors(DGE.all, method = "TMM")  # function from edgeR


# make a normalized counts matrix for use outside of limma (PCA, etc)
#print("Normalization factors for the first few libraries:")
#head(DGE.all$samples, n=20)
norm.factors <- DGE.all$samples[,"norm.factors"]
normCounts.all <- as.data.frame(t(t(DGE.all$counts)/norm.factors))
DGE.all <- DGEList(counts=normCounts.all)

# transform into logCPM counts first
lcpm.normCounts <- edgeR::cpm(DGE.all, log=TRUE, prior.count = 1)

## PCA to look at how the datasets differ
library(ggfortify)
pca_res <- prcomp(t(lcpm.normCounts),scale=TRUE) 
var_explained <- pca_res$sdev^2/sum(pca_res$sdev^2)
results <- as.data.frame(pca_res$x)
results$specimenID <- rownames(results)
anno <- NTAP_pub_rnaseq_data %>% 
    dplyr::select("specimenID", "tumorType", "studyName")
  
plot_data <- merge(results, anno)

plot_data %>% 
     ggplot(aes(x=PC1,
               y=PC2)) + 
    geom_point(aes(colour = as.factor(studyName), shape = as.factor(tumorType)), size=2) +
    theme_bw() + 
  labs(x=paste0("PC1: ",round(var_explained[1]*100,1),"%"),
       y=paste0("PC2: ",round(var_explained[2]*100,1),"%")) +
  theme(legend.position="right")


```

&nbsp;

## Differential gene expression analysis in samples:

We used limma-edgeR based analysis to find differentially expressed genes in the TMM normalized dataset. 

A point to note:

*  All MPNST samples except one are males while all PNF samples except one are femmales. So there may be a sex related effect on the analysis that we dont have a good way to mitigate. 
* The number of MPNST samples are extremely limited (n=4), so these results should be evaluated with more samples when possible

The plot below shows genes that are significantly overexpressed in MPNST compared to PNF or NF (logFC > 4) in red. The dots in blue refer to genes that are significantly underexpressed in MPNST compared to PNF or NF (logFC < -4). The thresholds of fold change have been arbitrarily chosen for ease of visualization in the volcano plot shown below.

### MPNST vs PNF

```{r DEG1, echo=F, eval= TRUE, results='hide', message=FALSE, warning=FALSE, fig.height=20, fig.width=20}

#Select annotations
annotes=NTAP_pub_rnaseq_data %>%
  dplyr::select(specimenID,sex,tumorType,studyName) %>%
  unique()
annotes$tumorType <- as.factor(annotes$tumorType)
annotes$sex <-as.factor(annotes$sex)
rownames(annotes)<-annotes$specimenID

## DEG analysis using limma, Glimma and EdgeR

limma_object <- DGE.all  # make new DGElist object so that we dont change the original object

#make design matrix
annotes <- annotes %>% 
  dplyr::filter(specimenID %in% colnames(gene_mat)) 
ordered_annotes <- annotes[match(colnames(gene_mat), annotes$specimenID),]
group_object <- ordered_annotes$tumorType
batch <- ordered_annotes$studyName
sex <- ordered_annotes$sex

design_new <- model.matrix(~0+group_object)
colnames(design_new) <- gsub("group_object", "", colnames(design_new)) # modify the colnames to remove "group_object" addition
colnames(design_new) <- gsub(" ", "_", colnames(design_new))
colnames(design_new) <- gsub("-", "_", colnames(design_new))
```

&nbsp;

#### DEG Analysis

```{r DEG_PNF, echo=F, eval= TRUE, results='hide', message=FALSE, warning=FALSE, fig.height=20, fig.width=20, fig.cap="Differential expression of genes in MPNST and pNF tumor types. Above shows a volcano plot highlighting significantly upregulated genes in red and downregulated genes in blue"}

contr.matrix <- makeContrasts(
   MPNSTvsPNF = Malignant_Peripheral_Nerve_Sheath_Tumor - Plexiform_Neurofibroma,
   levels = colnames(design_new))
#contr.matrix

## Using voom to make Elist object
#print("Visualizing the mean variance trend of the RNASeq dataset")
voom_object <- voomWithQualityWeights(limma_object, design_new, plot=FALSE) # removing heteroscadisticity
#voom_object$genes <- genes_unique

## Run DGE analysis using lmFit

fit_new <- lmFit(voom_object, design_new)
contrast_fit_new <- contrasts.fit(fit_new, contrasts=contr.matrix)
fit <- eBayes(contrast_fit_new, trend=TRUE)

results_fit_new <- decideTests(fit)
#summary(results_fit_new)

#vennDiagram(results_fit_new[,c(1)], circle.col=c("turquoise", "salmon"))
MPNSTvsPNF <- topTreat(fit, coef=1, n=Inf)
MPNSTvsPNF$Genes <- rownames(MPNSTvsPNF)
MPNSTvsPNF_ordered <- MPNSTvsPNF[order(MPNSTvsPNF$adj.P.Val),]

## Volcano Plots:

MPNSTvsPNF_ordered <-MPNSTvsPNF_ordered %>%
  mutate(threshold = ifelse(logFC >= 1 & adj.P.Val <= 0.05,"A", 
                            ifelse(logFC<=-1 & adj.P.Val <= 0.05, 
                                   "B", "C")))

#png(filename = "volcanoplot_mpnst_pnf.png", width = 1200, height = 1200)
ggplot(data=MPNSTvsPNF_ordered, aes(x=logFC, y=-log10(adj.P.Val))) +
  geom_point(aes(colour = threshold), alpha=0.3, size=8) +
  xlim(c(-10, 10)) + ylim(c(0, 2.5)) +
  xlab("log2 fold change") + ylab("-log10 adjusted Pvalue") +
  scale_colour_manual(values = c("A"= "red", "B"="blue",  "C"= "black")) +
  # geom_text(aes(label = ifelse(threshold != "C", 
  #                              as.character(Genes), ""), 
  #               colour = as.factor(threshold)), size = 10, angle = 45, hjust = -0.25) +
  theme_bw()+
  theme(legend.text = element_text(size=20), #element_text(size=8),
              axis.text.x  = element_text(size=40, angle = 0),
              axis.text.y = element_text(size=40),
              text = element_text(size=40),
              strip.text.x = element_text(size = 40),
              legend.position="none",
              panel.grid = element_blank(),
              panel.background = element_rect(fill = "white")) 
#dev.off()

annotes=NTAP_pub_rnaseq_data %>%
  dplyr::select(specimenID,sex,tumorType) %>%
  dplyr::filter(tumorType %in% c("Plexiform Neurofibroma", "Malignant Peripheral Nerve Sheath Tumor")) %>% 
  unique
rownames(annotes)<-annotes$specimenID

res <- lcpm.normCounts %>% 
  as.data.frame() %>% 
  dplyr::filter(rownames(lcpm.normCounts) %in% MPNSTvsPNF_ordered$Genes[1:100]) %>% 
  dplyr::select(matches(annotes$specimenID))


heatmap_1 <- pheatmap(res,
                       labels_col=rep("",ncol(res)),
                       fontsize_row = 10,
                       clustering_method = 'ward.D2',
                       annotation_col = dplyr::select(annotes, -specimenID),
                       treeheight_row = 25,
                       cutree_rows = 3,
                       width = 16, 
                       height = 8)
heatmap_1

plot(heatmap_1$tree_row, cex = 0.7)
abline(h=25, col="red", lty = 2, lwd = 2)
heatmap_cluster <- sort(cutree(heatmap_1$tree_row, h=25))

```

&nbsp;

#### Pathway analysis

```{r pathway_PNF, echo=F, eval= TRUE, fig.height=10, fig.width=10}


make_gprofiler_plots <- function(cluster_genes){
  
plot_df <- gprofiler2::gost(cluster_genes, organism = "hsapiens", 
  ordered_query = F, significant = T, exclude_iea = F,
  correction_method = "fdr", 
  domain_scope = "annotated")

t <- plot_df$result
terms_below_100 <- t[(t$term_size < 200 & t$source == "REAC" | t$term_size < 200 & t$source == "KEGG" | t$term_size < 200 & t$source == "`GO:CC`" | t$term_size < 200 & t$source == "`GO:BP`" | t$term_size < 200 & t$source == "WP"),]

plot_manhattan <- gprofiler2::gostplot(plot_df,
                     capped = FALSE,
                     interactive = F,
                     pal = c(`GO:MF` = "#dc3912", 
                             `GO:BP` = "#ff9900", 
                             `GO:CC` = "#109618", 
                             KEGG ="#dd4477", 
                             REAC = "#3366cc", 
                             WP = "#0099c6", 
                             TF = "#5574a6", 
                             MIRNA = "#22aa99", 
                             HPA = "#6633cc", 
                             CORUM = "#66aa00", 
                             HP = "#990099"
                             ))


library(DT)
table <- DT::datatable(terms_below_100[, c("source","term_name", "term_size", "query_size", "intersection_size", "p_value")])

my_list <- list(plot_manhattan, table)
return(my_list)
  
}


cluster_gene_df <- as.data.frame(heatmap_cluster)
cluster_gene_df$gene <- row.names(cluster_gene_df)

print("Enrichment Plot of Cluster 1 genes")
output <- make_gprofiler_plots(cluster_gene_df$gene[cluster_gene_df$heatmap_cluster == 1])

output[[1]]
output[[2]]


print("Enrichment Plot of Cluster 2 genes")
output <- make_gprofiler_plots(cluster_gene_df$gene[cluster_gene_df$heatmap_cluster == 2])

output[[1]]
output[[2]]


print("Enrichment Plot of Cluster 3 genes")
output <- make_gprofiler_plots(cluster_gene_df$gene[cluster_gene_df$heatmap_cluster == 3])

output[[1]]
output[[2]]

print("List of genes in Cluster 3")
cluster_gene_df$gene[cluster_gene_df$heatmap_cluster == 3]

```

&nbsp;

### MPNST vs NF

#### DEG Analysis

```{r DEG_NF, echo=F, eval= TRUE, results='hide', message=FALSE, warning=FALSE, fig.height=20, fig.width=20, fig.cap="Differential expression of genes in MPNST and NF tumor types. Above shows a volcano plot highlighting significantly upregulated genes in red and downregulated genes in blue"}

contr.matrix_1 <- makeContrasts(
   MPNSTvsNF = Malignant_Peripheral_Nerve_Sheath_Tumor - Neurofibroma,
   levels = colnames(design_new))

## Using voom to make Elist object
#print("Visualizing the mean variance trend of the RNASeq dataset")
voom_object <- voomWithQualityWeights(limma_object, design_new, plot=FALSE) # removing heteroscadisticity
#voom_object$genes <- genes_unique

## Run DGE analysis using lmFit

fit_new <- lmFit(voom_object, design_new)
contrast_fit_new <- contrasts.fit(fit_new, contrasts=contr.matrix_1)
fit <- eBayes(contrast_fit_new, trend=TRUE)

results_fit_new <- decideTests(fit)
#summary(results_fit_new)

#vennDiagram(results_fit_new[,c(1)], circle.col=c("turquoise", "salmon"))
MPNSTvsNF <- topTreat(fit, coef=1, n=Inf)
MPNSTvsNF$Genes <- rownames(MPNSTvsNF)
MPNSTvsNF_ordered <- MPNSTvsNF[order(MPNSTvsNF$adj.P.Val),]

## Volcano Plots:

MPNSTvsNF_ordered <-MPNSTvsNF_ordered %>%
  dplyr::mutate(threshold = ifelse(logFC >= 1 & adj.P.Val <= 0.05,"A", 
                            ifelse(logFC<=-1 & adj.P.Val <= 0.05, 
                                   "B", "C")))

#png(filename = "volcanoplot_mpnst_pnf.png", width = 1200, height = 1200)
ggplot(data=MPNSTvsNF_ordered, aes(x=logFC, y=-log10(adj.P.Val))) +
  geom_point(aes(colour = threshold), alpha=0.3, size=8) +
  xlim(c(-10, 10)) + ylim(c(0, 2.5)) +
  xlab("log2 fold change") + ylab("-log10 adjusted Pvalue") +
  scale_colour_manual(values = c("A"= "red", "B"="blue",  "C"= "black")) +
  # geom_text(aes(label = ifelse(threshold != "C", 
  #                              as.character(Genes), ""), 
  #               colour = as.factor(threshold)), size = 10, angle = 45, hjust = -0.25) +
  theme_bw()+
  theme(legend.text = element_text(size=20), #element_text(size=8),
              axis.text.x  = element_text(size=40, angle = 0),
              axis.text.y = element_text(size=40),
              text = element_text(size=40),
              strip.text.x = element_text(size = 40),
              legend.position="none",
              panel.grid = element_blank(),
              panel.background = element_rect(fill = "white")) 
#dev.off()

annotes=NTAP_pub_rnaseq_data %>%
  dplyr::select(specimenID,sex,tumorType) %>%
  dplyr::filter(tumorType %in% c("Neurofibroma", "Malignant Peripheral Nerve Sheath Tumor")) %>% 
  unique
rownames(annotes)<-annotes$specimenID

res_NF <- lcpm.normCounts %>% 
  as.data.frame() %>% 
  dplyr::filter(rownames(lcpm.normCounts) %in% MPNSTvsNF_ordered$Genes[1:100]) %>% 
  dplyr::select(matches(annotes$specimenID))


heatmap_2 <- pheatmap(res_NF,
                       labels_col=rep("",ncol(res)),
                       fontsize_row = 10,
                       clustering_method = 'ward.D2',
                       annotation_col = dplyr::select(annotes, -specimenID),
                       treeheight_row = 25,
                       cutree_rows = 4,
                       width = 16, 
                       height = 8)
heatmap_2

plot(heatmap_2$tree_row, cex = 0.7)
abline(h=25, col="red", lty = 2, lwd = 2)
heatmap_cluster2 <- sort(cutree(heatmap_2$tree_row, h=25))

```

&nbsp;

#### Pathway analysis

```{r pathway_NF, echo=F, eval= TRUE, fig.height=10, fig.width=10}

cluster_gene_df <- as.data.frame(heatmap_cluster2)
cluster_gene_df$gene <- row.names(cluster_gene_df)

print("Enrichment Plot of Cluster 1 genes")
output <- make_gprofiler_plots(cluster_gene_df$gene[cluster_gene_df$heatmap_cluster2 == 1])

output[[1]]
output[[2]]


print("Enrichment Plot of Cluster 2 genes")
output <- make_gprofiler_plots(cluster_gene_df$gene[cluster_gene_df$heatmap_cluster2 == 2])

output[[1]]
output[[2]]

print("List of genes in Cluster 2 (cluster with PLK1)")
cluster_gene_df$gene[cluster_gene_df$heatmap_cluster2 == 2]


print("Enrichment Plot of Cluster 3 genes")
output <- make_gprofiler_plots(cluster_gene_df$gene[cluster_gene_df$heatmap_cluster2 == 3])

output[[1]]
output[[2]]

print("Enrichment Plot of Cluster 4 genes")
output <- make_gprofiler_plots(cluster_gene_df$gene[cluster_gene_df$heatmap_cluster2 == 4])

output[[1]]
output[[2]]



```

&nbsp;
